src/include/timestamp_inline.h

```diff
@@ -199,3 +199,29 @@
         if ((source)->prepare != 0)                                                           \
             (dest)->prepare = 1;                                                              \
     } while (0)
+
+/* Abstract away checking whether all records in an aggregated time window have been deleted. */
+#define WT_TIME_AGGREGATE_ALL_DELETED(ta) (ta)->newest_stop_ts != WT_TS_MAX
+
+/*
+ * Update a time aggregate in preparation for an obsolete visibility check. This deserves a macro,
+ * since the mechanism for identifying whether an aggregated time window contains only obsolete (i.e
+ * deleted) data requires checking two different timestamps. Note the output time aggregate might be
+ * either empty initialized, or have been populated via prior calls to this macro with other
+ * aggregated windows.
+ */
+#define WT_TIME_AGGREGATE_MERGE_OBSOLETE_VISIBLE(session, out_ta, in_ta)                         \
+    do {                                                                                         \
+        WT_ASSERT(session, (out_ta)->init_merge == 1);                                           \
+        (out_ta)->newest_stop_durable_ts =                                                       \
+          WT_MAX((out_ta)->newest_stop_durable_ts, (in_ta)->newest_stop_durable_ts);             \
+        (out_ta)->newest_txn = WT_MAX((out_ta)->newest_txn, (in_ta)->newest_txn);                \
+        /*                                                                                       \
+         * Ensure that visibility checks return false if there is any content that               \
+         * has not been deleted.                                                                 \
+         */                                                                                      \
+        if (WT_TIME_AGGREGATE_ALL_DELETED((in_ta)))                                              \
```

Comment:
If we are checking outside whether the `newest_stop_ts` is WT_TS_MAX or not, then this check for setting MAX is not required as it is automatically calculated into the `newest_stop_ts` because of WT_MAX().

If we do not check this condition outside, then the condition check must be `if (!WT_TIME_AGGREGATE_ALL_DELETED())`. If all the values are not deleted, we set the durable timestamp `newest_stop_durable_ts` as WT_TS_MAX and use this timestamp directly outside for visibility check.

Reply:
Thanks Hari - that makes sense. I struggled to get clarity last night. I've pushed up a different version that copies WT_TS_MAX into the durable stop timestamp. I'm not convinced that's a good idea, but it does successfully abstract out the ugliness.